src/main/java/seedu/mtracker/console/AddInstrumentParser.java

```diff
@@ -34,11 +35,19 @@ public static String getInstrumentNameFromUser(String instrumentType) {
     public static boolean isValidName(String name, String instrumentType) {
         boolean isValid = true;
         try {
-            if (name.isEmpty()) {
+            if (instrumentType.equals(AddForexParser.INSTRUMENT_TYPE)) {
+                if (name.length() != FX_PAIR_NAME_LENGTH) {
+                    throw new IllegalArgumentException();
+                }
+            } else if (name.isEmpty()) {

```

Comment:
```suggestion
            if (name.isEmpty() || (instrumentType.equals(AddForexParser.INSTRUMENT_TYPE) && name.length() != FX_PAIR_NAME_LENGTH)) {
               throw new IllegalArgumentException();
```

Simplify the conditional, to make the code better maybe abstract out the conditional here